src/MonitoringFunctions.Common/HelperMethods.cs

```diff
@@ -19,28 +19,67 @@ internal static class HelperMethods
         private static readonly HttpClient _httpClient = new HttpClient();
 
         /// <summary>
-        /// Tests if the contents of the given url can be accessed from current environment.
-        /// Saves results to Kusto.
+        /// Tests if the contents of the given url can be accessed from the current environment.
+        /// Reports results to given data service.
         /// </summary>
         /// <param name="log"><see cref="ILogger"/> that is used to report log information.</param>
         /// <param name="monitorName">Name of this monitor to be included in the logs and in the data sent to Kusto.</param>
         /// <param name="url">Url that this method will attempt to access.</param>
+        /// <param name="dataService">Data service to be used when reporting the results.</param>
+        /// <param name="cancellationToken">Token to cancel the asynchronous operation.</param>
         /// <returns>A task, tracking the initiated async operation. Errors should be reported through exceptions.</returns>
-        internal static async Task CheckAndReportUrlAccessAsync(ILogger log, string monitorName, string url, IDataService dataService,
+        internal static async Task CheckAndReportUrlAccessAsync(ILogger log,
+            string monitorName,
+            string url,
+            IDataService dataService,
             CancellationToken cancellationToken = default)
         {
-            HttpResponseMessage response = await _httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
+            _ = log ?? throw new ArgumentNullException(paramName: nameof(log));
+            _ = string.IsNullOrWhiteSpace(monitorName) ? throw new ArgumentNullException(paramName: nameof(monitorName)) : monitorName;
+            _ = string.IsNullOrWhiteSpace(url) ? throw new ArgumentNullException(paramName: nameof(url)) : url;
+            _ = dataService ?? throw new ArgumentNullException(paramName: nameof(dataService));
 
-            await dataService.ReportUrlAccessAsync(monitorName, response, cancellationToken).ConfigureAwait(false);
-
-            if (response.IsSuccessStatusCode)
+            HttpRequestLogEntry logEntry = new HttpRequestLogEntry()
+            {
+                MonitorName = monitorName,
+                EventTime = DateTime.UtcNow,
+                RequestedUrl = url
+            };
+            HttpResponseMessage? response = null;
+            try
             {
-                log.LogInformation($"Monitor '{monitorName}' has succeeded accessing the url {url}");
+                response = await _httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
+                logEntry.HttpResponseCode = (int)response.StatusCode;
+                await dataService.ReportUrlAccessAsync(logEntry, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception httpException)
```

Comment:
if you handle [HttpRequestException](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestexception?view=net-5.0) separately you might get extra details beneficial for log, for example status code received.

Reply:
If there is a StatusCode, we are already acquiring it in the `response` returned by the `httpClient.GetAsync(...)` method. The method throws if there is an error accessing the server. So, if we arrive in this catch block, there won't be a status code to log.

Reply:
`httpClient.GetAsync(...)` throws `HttpRequestException` for invalid status codes, for example 404. Then received response code will be available in `HttpRequestException.StatusCode` property.

Reply:
MSDN says that GetAsync() can throw HttpRequestException, however I'm not sure if any of cases might be caused by unsuccessful response - I haven't found one. Probably it neither of it would contain the status code.

Reply:
Are you actually taking care of whether response is successful or not in this scenario?

Reply:
MSDN mentions that an `HttpRequestException` is thrown if the request failed. If we get, let's say, `Http 500 - Internal Server Error` this means that the request succeeded. The server, processed our request and successfully sent us a response. In this case, an exception wouldn't be thrown (my tests with httpstat.us confirm this).

As far as I understand, response code in `HttpRequestException` only makes sense if the exception is thrown from `HttpResponseMessage.IsSuccessStatusCode`.

As for your question, we don't care much if the response is successful or not. We just send the value to the database.